sdk/core/azure-core/azure/core/serialization.py

```diff
@@ -95,29 +107,16 @@ class AzureJSONEncoder(JSONEncoder):
     """A JSON encoder that's capable of serializing datetime objects and bytes."""
 
     def default(self, o):  # pylint: disable=too-many-return-statements
+        if isinstance(o, (bytes, bytearray)):
+            return base64.b64encode(o).decode()
+        try:
```

Comment:
nit: to make it more clear what order we're trying things in inside the method body:
```suggestion
        try:
            # First, try datetime.datetime and datetime.date/datetime.time
```